synapse/storage/databases/main/events_bg_updates.py

```diff
@@ -715,142 +736,37 @@ def get_rejected_events(
     async def _chain_cover_index(self, progress: dict, batch_size: int) -> int:
         """A background updates that iterates over all rooms and generates the
         chain cover index for them.
+
+        Returns:
+            tuple of rows processed, if the background update has finished and
+            the new progress dict.
```

Comment:
I'm having some trouble following what this is trying to say. (Also I think the type hint needs updating?)

Reply:
Oh argh, that is a holdover from a previous attempt to make this better :facepalm: